feat(docs): Describe coordinate systems and reference points (#510) (#516)
OpenSimulationInterface/open-simulation-interface: doc/architecture/reference_points_coordinate_systems.adoc

= Reference points and coordinate systems

> With Open Simulation Interface, an object's position coordinates refer to the center of the object's bounding box, independent of the coordinate system. This reference point is not identical to the origin of any coordinate systems.

Reviewer: "...independent of the coordinate system." -> can this be dropped? 

I have a hard time to understand the second sentence. Do you mean: "The reference point does not have to be identical to the origin of sensor coordinates or object coordinates"? 

Would it make sense to shift the two sentences under "World coordinates"?

OSI uses three coordinate systems:

World coordinates::
Coordinate system for all quantities which are part of ground truth.

Sensor coordinates::
Coordinate system for all quantities which are part of sensor data.

Object coordinates::
Coordinate system for local object coordinates, for example, vehicle coordinates.

> 

Reviewer: For all coordinate systems: Would it make sense to say something about the origin? e.g.
World coordinates: inertial x/y/z positions
Sensor coordinates: mounting position of the sensor / virtual mounting position
Object coordinates: For vehicles it is currently middle rare axle, but there were discussions about maybe changing it to middle bounding box

Reviewer 2: **Output CCB 09.06.2021:**
1. Describe OSI's Coordinate systems in a generic manner.
2. Describe how to define the position of the coordinate systems in OSI (for example wrt reference points, which may be the center of the bounding box).
3. Example picture to illustrate the above mentioned points (a picture is already available needs to be updated).

= Reference points and coordinate systems

> With Open Simulation Interface, an object's position coordinates refer to the center of the object's bounding box.

Reviewer 2: ```suggestion
In Open Simulation Interface, object's coordinates refer to the position of its center of bounding box.
```

This reference point is not necessarily identical to the origin of the respective coordinate systems.
OSI uses DIN ISO 8855:2013-11 for coordinate systems and transformations between coordinate systems.

OSI uses three coordinate systems:

World coordinate system::
Coordinate system for all quantities which are part of ground truth.
The world coordinate system is an inertial x/y/z-coordinate system.

Sensor coordinate system::
Coordinate system for all quantities which are part of sensor data.
The origin is the mounting position of the physical sensor or a virtual mounting position, depending on the technology.

Object coordinate system::
Coordinate system for local object coordinates, for example, vehicle coordinates.
For vehicles, the origin is the middle of the rear axle projected onto the ground.
Other objects may use different origins.

> The following image shows an example of the differents coordinate systems.

Reviewer 2: ```suggestion
The following image shows an example of the coordinate systems.
```

It also shows that the reference point of the vehicle is located in the center of its bounding box whereas the origin of the vehicle coordinate system is the middle of rear axle projected onto the ground.
In this example, the virtual mounting position coincides with the origin of the vehicle coordinate system.

> image::{imagedir}/osi_coordinate_systems.png[]

Reviewer 2: We should use a better image for this, one that does not cover the "Z" on the virtual coordinate system and that also shows the virtual coordinate system in a different location from the rear axle to evade confusion with the DIN ISO.

Author: Simplified the image and moved the example to the topic describing the transformations between coordinate systems.
Seemed to me to be the more appropriate place.
